internal/pkg/admission/policy-server-ca-secret.go

```diff
@@ -97,27 +128,59 @@ func extractCaFromSecret(caSecret *corev1.Secret) (*admissionregistration.CA, er
 	return &admissionregistration.CA{CaCert: caCert, CaPrivateKey: caPrivateKey}, nil
 }
 
-func (r *Reconciler) fetchOrInitializePolicyServerCARootSecret(ctx context.Context, generateCA generateCAFunc, pemEncodeCertificate pemEncodeCertificateFunc) (*corev1.Secret, error) {
-	policyServerSecret := corev1.Secret{}
+// FetchOrInitializeRootCASecret checks if the root CA secret exists and
+// creates it if it does not.  It returns the secret containing the root CA, a
+// boolean flag to signal if the secret has been initialize now and an error in
+// case of failure.
+func (r *Reconciler) FetchOrInitializeRootCASecret(ctx context.Context, generateCA generateCAFunc, pemEncodeCertificate pemEncodeCertificateFunc) (*corev1.Secret, bool, error) {
+	CASecret := corev1.Secret{}
 	err := r.Client.Get(
 		ctx,
 		client.ObjectKey{
 			Namespace: r.DeploymentsNamespace,
-			Name:      constants.PolicyServerCARootSecretName},
-		&policyServerSecret)
+			Name:      constants.KubewardenCARootSecretName},
+		&CASecret)
 	if err != nil && apierrors.IsNotFound(err) {
-		return r.buildPolicyServerCARootSecret(generateCA, pemEncodeCertificate)
+		CASecret, err := r.buildRootCASecret(generateCA, pemEncodeCertificate)
+		if err != nil {
+			return &corev1.Secret{}, false, err
+		}
+		if err := r.reconcileCASecret(ctx, CASecret); err != nil {
+			return &corev1.Secret{}, false, err
+		}
+		return CASecret, true, nil
 	}
-	policyServerSecret.ResourceVersion = ""
+
+	CASecret.ResourceVersion = ""
 	if err != nil {
-		return &corev1.Secret{},
+		return &corev1.Secret{}, false,
 			fmt.Errorf("cannot fetch or initialize Policy Server CA secret: %w", err)
 	}
+	return &CASecret, false, nil
+}
+
+// FetchKubewardenCARootSecret function used to get the secret storing the root
+// CA secret used by Kubewarden controllers.
+func (r *Reconciler) FetchKubewardenCARootSecret(ctx context.Context) (*corev1.Secret, error) {
```

Comment:
It seems pretty similar to what `FetchOrInitializeRootCASecret`, do we need both methods?